app/models/project.js

```diff
@@ -31,8 +41,80 @@ export default class ProjectModel extends Model {
 
   @hasMany('milestone', { async: false }) milestones;
 
+  // Attributes for dashboard view
+  @attr() tab;
+
+  @attr() teammemberrole;
+
   @attr() applicantteam;
 
+  @computed('milestones')
+  get publicReviewPlannedStartDate() {
+    const cbMilestone = this.milestones.find(milestone => milestone.milestone === COMMUNITY_BOARD_REFERRAL_MILESTONE_ID);
+    if (cbMilestone) {
+      return cbMilestone.plannedstartdate;
+    }
+    return null;
+  }
+
+  // These displayMilestoneActual<Start/End>
+  // date fields represent different dates from different milestones
+  // depending on `tab`.
+  // If `tab` is 'upcoming' or 'to-review', these dates are derived from
+  // the participant's (specified by `teammemberrole`) review milestone.
+  // If `tab` is 'reviewed'...
+  //   - these dates come from the current In Progress milestone
+  //   - if there is two In Progress milestones, an array of milestone dates is returned
+  @computed('tab', 'teammemberrole', 'milestones')
+  get displayMilestoneActualStartDate() {
+    if (this.tab === 'to-review') {
+      const participantMilestoneId = MILESTONE_ID_LOOKUP[this.teammemberrole];
+      const participantReviewMilestone = this.milestones.find(milestone => milestone.milestone === participantMilestoneId);
+      return participantReviewMilestone.actualstartdate;
+    }
+    if (this.tab === 'reviewed') {
+      const inProgressMilestones = this.milestones.filter(milestone => milestone.statuscode === 'In Progress');
+      return inProgressMilestones.map(milestone => ({
+        milestone: milestone.milestone,
+        actualstartdate: milestone.actualstartdate,
+      }));
+    }
+    return null;
+  }
+
+  @computed('tab', 'teammemberrole', 'milestones')
+  get displayMilestoneActualEndDate() {
+    if (this.tab === 'to-review') {
+      const participantMilestoneId = MILESTONE_ID_LOOKUP[this.teammemberrole];
+      const participantReviewMilestone = this.milestones.find(milestone => milestone.milestone === participantMilestoneId);
+      return participantReviewMilestone.actualenddate;
+    }
+    if (this.tab === 'reviewed') {
+      const inProgressMilestones = this.milestones.filter(milestone => milestone.statuscode === 'In Progress');
+      return inProgressMilestones.map(milestone => ({
+        milestone: milestone.milestone,
+        actualenddate: milestone.actualenddate,
+      }));
+    }
+    return null;
+  }
```

Comment:
Yeah, similar to my comment above, we have a lot of these "display" type computed properties. Are they used only in a specific component? What do you think about this: distill from here the actual raw data that informs those views AND the component-specific data structures (for example, this `map` into an array of objects... is that a data structure that's expected by a component somewhere? These seem to know too much about each other...

Reply:
I like this idea a lot. Makes sense -- breaking the computed down into two tab-specific computeds, one for "to-review" and another for "reviewed", and tying each computed to a tab-specific component.